setup.py

```diff
@@ -1,308 +1,114 @@
-
+        self.fast = False
+        self.gcov = False
+        self.vector = False
+        self.mpir = False
+        self.static = False
+        self.gdb = False
 
     def finalize_options(self):
         build_ext.finalize_options(self)
-        gmpy_build_ext.doit(self)
-
-# custom install data in order that data_files
-# get installed together with the .so
-
-class gmpy_install_data(install_data):
-    def finalize_options(self):
-        install_data.finalize_options(self)
-        install = self.distribution.get_command_obj('install')
-        self.install_dir = install.install_purelib
-
-# prepare the extension for building
-
-my_commands = {'clean' : gmpy_clean, 'build_ext' : gmpy_build_ext, 'install_data' : gmpy_install_data}
-
-gmpy2_ext = Extension('gmpy2',
-                      sources=[os.path.join('src', 'gmpy2.c')],
-                      include_dirs=['./src'],
-                      define_macros = defines)
-
-setup(name = "gmpy2",
-      version = "2.1.0a1",
-      author = "Case Van Horsen",
-      author_email = "casevh@gmail.com",
-      license = "LGPL-3.0+",
-      url = "https://github.com/aleaxit/gmpy",
-      description = "gmpy2 interface to GMP/MPIR, MPFR, and MPC for Python 2.6+ and 3.4+",
-      data_files = [('', ['src/gmpy2.pxd']), ('gmpy2', ['src/gmpy2.h'])],
-      classifiers = [
+        if self.fast:
+            _comp_args.append('-DFAST=1')
+        if self.gcov:
+            if ON_WINDOWS:
+                raise ValueError("Cannot enable GCC code coverage on Windows")
+            _comp_args.append('-DGCOV=1')
+            _comp_args.append('-O0')
+            _comp_args.append('--coverage')
+            link_args.append('--coverage')
+            _libs.append('gcov')
+        if self.vector:
+            _comp_args.append('-DVECTOR=1')
+        if self.static and not ON_WINDOWS:
+            _comp_args.remove('-DSHARED=1')
+            _comp_args.append('-DSTATIC=1')
+        if self.gdb:
+            _comp_args.append('-ggdb')
+
+    def build_extensions(self):
+        compiler = self.compiler.compiler_type
+        if compiler == 'mingw32':
+            _comp_args.append('-DMSYS2=1')
+            if self.mpir:
+                _comp_args.append('-DMPIR=1')
+                _libs.append('libmpir')
+            else:
+                # Should library names include 'lib' on mingw32/msys2?
+                _libs.append('libgmp')
+        elif self.mpir or ON_WINDOWS:
+            # --mpir or on Windows and MSVC
+            _comp_args.append('-DMPIR=1')
+            if ON_WINDOWS:
+                _libs.append('libmpir')
+            else:
+                _libs.append('mpir')
+        else:
+            _libs.append('gmp')
+        build_ext.build_extensions(self)
+
+
+extensions = [
+    Extension('gmpy2.gmpy2',
+              sources=sources,
+              include_dirs=["/usr/local/include", './src'],
```

Comment:
Please remove "/usr/local/include"

Reply:
Why? User installed libraries will go in there and unless included here, setuptools will not find those libraries.

Reply:
If user installed libraries go there, it should be given with `-I` to build_ext.
Problem with this is that, /usr/local/include is searched first, which makes it impossible to build with another installation if there are multiple installations.

Reply:
That's not accurate. `include_dirs` takes a list of paths. Setuptools itself produces the `-I` directives to the compiler (and the directive depends on the compiler). You can see the build output by running `python setup.py build`.

Yes, with this `/usr/local` will come first. It can always be overriden with the `--include-dirs (-I)` built in setuptools option. See `python setup.py build_ext -h`.

Reply:
> Yes, with this /usr/local will come first. It can always be overriden with the --include-dirs (-I) built in setuptools option to allow for multiple installations. See python setup.py build_ext -h.

Here is what I get,
```
isuru@dell:~/projects/gmpy$ python setup.py build_ext --vector --fast --include-dirs=qwe
running build_ext
building 'gmpy2.gmpy2' extension
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security -fPIC -I/usr/local/include -I./src -Iqwe -I/usr/include/python2.7 -c src/gmpy2.c -o build/temp.linux-x86_64-2.7/src/gmpy2.o -DSHARED=1 -DFAST=1 -DVECTOR=1
```

Reply:
Well, `/usr/local` is typically included in system paths by modern versions of gcc. Setuptools seems to append dirs specified in `--include-dirs` rather than override or add them first. So it can be removed. Worst case scenario users will have to add `/usr/local` via `include-dirs`.